### PR TITLE
RHIDP-1166: Create a software template for a Backend plugin

### PR DESCRIPTION
**Pull request by jesuino**
## What does this PR do / why we need it

This PR creates a software template to scaffold a backend plugin skeleton.

Users can either push the plugin skeleton to a new repository or create a PR in an existing one.

I followed same template proposed by Debsmita on [this PR](https://github.com/redhat-developer/red-hat-developer-hub-software-templates/pull/5).

It also includes some techdocs, more can be added later if required.

A sample [Plugin PR](https://github.com/jesuino/rhdh-sample-plugins/pull/2) was created from this template and it works on my existing Backstage Repo.

## Which issue(s) does this PR fix

[RHIDP-1166](https://issues.redhat.com/browse/RHIDP-1166)

## How to test changes / Special notes to the reviewer

Import the `template.yaml` on a RHDH 1.2 installation and run it with the required parameters.

### CODE REVIEW

**Comment by jesuino** [comment]
@gashcrumb
@debsmita1 

Here's my backend plugin software template. Comments and suggestions are welcome. Thanks!

File: templates/create-backend-plugin/docs/features/export_to_rhdh.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+global:
+  dynamic:
+    plugins:
+      - package: path to your plugin, or @npmPackage@version
+        disabled: false
+        # this is required if not using a package in filesystem
+        integrity: sha512-...
+```
+
+For more informatino check [Dynamic Plugins](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/dynamic-plugins.md) documentation.
```
Comment (gashcrumb) [inline templates/create-backend-plugin/docs/features/export_to_rhdh.md:29]: typo here, `information`

File: templates/create-backend-plugin/docs/features/export_to_rhdh.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+```json
+"export-dynamic": "janus-cli package export-dynamic-plugin --clean"
+```
+
+
+## Installation
+
+After running this command a new folder will be created in root dir: `dist-dynamic`. This folder should be copied to RHDH dynamic plugins root directory and renamed to `${plugin}-dynamic` or it is possible to publish its content to a NPM registry. The plugin can be mapped on `app-config.yaml` using the dynamic section, for example:
```
Comment (gashcrumb) [inline templates/create-backend-plugin/docs/features/export_to_rhdh.md:17]: I don't think we want to mention publishing to NPM registries anymore.  Maybe also it'd be worth in the short term adding a disclaimer that this section/doc is referring to tech preview functionality.

**Comment by jesuino** [comment]
Thanks for your review, @gashcrumb changes were made, let me know if I am missing anything.

Thanks!

File: templates/create-backend-plugin/template.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: create-backend-plugin
+  title: Create Backend Plugin Template
```
Comment (debsmita1) [inline templates/create-backend-plugin/template.yaml:5]: Add this annotation to load tech docs

```
  annotations:
    backstage.io/techdocs-ref: dir:.
```

**Comment by gashcrumb** [comment]
Gave this a test here and looks great!  I like the additional endpoint you added as well.  I agree with @debsmita1 probably it'd be good to add the tech docs annotation by default after which I think this PR is ready to be merged.

File: templates/create-backend-plugin/template.yaml (hunk 3 above)
Comment (jesuino) [reply templates/create-backend-plugin/template.yaml:5]: Thank you for this, just added the techdocs annotation!

**Review by gashcrumb — APPROVED** [review]

**Pull request merged**